src/gadgets/mpt_update.rs

```diff
@@ -20,7 +21,7 @@ use strum_macros::EnumIter;
 
 /// Each row of an mpt update belongs to one of four segments.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, EnumIter)]
-enum SegmentType {
+pub(crate) enum SegmentType {
```

Comment:
We can just make this pub and not re-export it at the lib level. I learned this from Aurel on slack:
"Fun fact: most  pub(crate)  everywhere do nothing. It is a syntax for special cases. The basic way to expose an item to parent and sibling modules is to mark it pub and keep the module family under a common non-pub module (util in this example; (crate) means nothing)."

Reply:
Replace `pub(crate)` with `pub` for `SegmentType` in commit [8a2095f](https://github.com/scroll-tech/mpt-circuit/pull/32/commits/8a2095f54891d1fa8e7a5de3e093a9012e6bb16c). Thanks.